src/main/java/org/veupathdb/service/eda/ds/plugin/pass/MapMarkersOverlayPlugin.java

```diff
@@ -14,94 +14,109 @@
-    PieplotSpec spec = getPluginSpec();
+    MapMarkersOverlaySpec spec = getPluginSpec();
     String showMissingness = spec.getShowMissingness() != null ? spec.getShowMissingness().getValue() : "noVariables";
     String deprecatedShowMissingness = showMissingness.equals("FALSE") ? "noVariables" : showMissingness.equals("TRUE") ? "strataVariables" : showMissingness;
     String valueSpec = singleQuote(spec.getValueSpec().getValue());
+    String xVarType = getVariableType(spec.getXAxisVariable());
 
     Map<String, VariableSpec> varMap = new HashMap<String, VariableSpec>();
     varMap.put("xAxisVariable", spec.getXAxisVariable());
-    varMap.put("facetVariable1", getVariableSpecFromList(spec.getFacetVariable(), 0));
-    varMap.put("facetVariable2", getVariableSpecFromList(spec.getFacetVariable(), 1));
+    varMap.put("geoAggregateVariable", spec.getGeoAggregateVariable());
     varMap.put("latitudeVariable", spec.getLatitudeVariable());
     varMap.put("longitudeVariable", spec.getLongitudeVariable());
       
     useRConnectionWithRemoteFiles(dataStreams, connection -> {
       connection.voidEval(getVoidEvalFreadCommand(DEFAULT_SINGLE_STREAM_NAME, 
           spec.getXAxisVariable(),
-          getVariableSpecFromList(spec.getFacetVariable(), 0),
-          getVariableSpecFromList(spec.getFacetVariable(), 1),
+          spec.getGeoAggregateVariable(),
           spec.getLatitudeVariable(),
           spec.getLongitudeVariable()));
       connection.voidEval(getVoidEvalVarMetadataMap(DEFAULT_SINGLE_STREAM_NAME, varMap));
 
       String viewportRString = getViewportAsRString(spec.getViewport());
       connection.voidEval(viewportRString);
       
+      String binReportValue = "NULL";
+      if (xVarType.equals("STRING")) {
+        // maybe i should have a warning here if they pass a BinSpec?
+        connection.voidEval("binWidth <- NULL");
+      } else {
+        BinSpec binSpec = spec.getBinSpec();
+        validateBinSpec(binSpec, xVarType);
+        binReportValue = binSpec.getType().getValue() != null ? binSpec.getType().getValue() : "binWidth";
+        String binWidth = connection.eval("numBinsToBinWidth(" + DEFAULT_SINGLE_STREAM_NAME + "[[map$variableId[map$plotRef == 'xAxisVariable']]], 8)").toString();
+        if (xVarType.equals("NUMBER") || xVarType.equals("INTEGER")) {
+          binWidth = binSpec.getValue() == null ? binWidth : "as.numeric('" + binSpec.getValue() + "')";
+        } else {
+          binWidth = binSpec.getValue() == null ? binWidth : "'" + binSpec.getValue().toString() + " " + binSpec.getUnits().toString().toLowerCase() + "'";
+        }
+        connection.voidEval("binWidth <- " + binWidth);
+      }
+
       String cmd =
-          "plot.data::pie(" + DEFAULT_SINGLE_STREAM_NAME + ", map, " +
-              valueSpec + ", viewport, '" +
+          "plot.data::mapMarkers(" + DEFAULT_SINGLE_STREAM_NAME + ", map, binWidth, " +
```

Comment:
plot.data::mapMarkers is still pending a push to plot.data repo?

Reply:
see the recent veupathdb/plot.data#166